### ISSUE TITLE
[Bug]: When joint_strategy is rear in RingAttention, torch.cat should be performed in the final step.

### ISSUE THREAD

**Issue by iwzbi** [vllm-project/vllm-omni#673]
### Your current environment

empty

### 🐛 Describe the bug

In my understanding, when `joint_query` has already been placed at the end of the tensor `query`,
https://github.com/vllm-project/vllm-omni/blob/14c04d0fd4d9f1dfd5370c481a795885565668e4/vllm_omni/diffusion/attention/parallel/ring.py#L81-L84
extra caution should be taken with the `causal=True` case to ensure consistency in the order of query and key/value tokens.
https://github.com/vllm-project/vllm-omni/blob/14c04d0fd4d9f1dfd5370c481a795885565668e4/vllm_omni/diffusion/attention/backends/ring_pytorch_attn.py#L87-L101
In the code above, `joint_tensor_key` is concatenated with `step_k` on the sp rank in step 0, which leads to a different token order for the overall key and query. If `causal=is_causal and step == 0` evaluates to True, the final attention result should be incorrect.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.

**Comment by david6666666** [comment]
cc @mxuax @wtomin

**Comment by mxuax** [comment]
Your understanding is correct. 
When joint_strategy="rear" specific attention-mask should be designed instead of  is_causal=True. But the majority of existing models chose the "front" strategy. Currently self-designed attention mask is not supported in SP, but we are working on this, I think.
